src/blame.c

```diff
@@ -451,7 +455,12 @@ blame_request(struct view *view, enum request request, struct line *line)
 			if (diff->pipe)
 				string_copy_rev(diff->ref, NULL_ID);
 		} else {
-			open_diff_view(view, flags);
+			if (filename_from_blame) {
+				free(filename_from_blame);
+			}
+			filename_from_blame = strdup(state->filename);
```

Comment:
This does not work correctly if the file was renamed since the blamed line. TODO: fix (use filename from blame->commit), test.